gnuradio-runtime/lib/hier_block2_detail.cc

```diff
@@ -996,5 +1017,70 @@ void hier_block2_detail::reset_hier_blocks_parent()
     }
 }
 
+std::set<basic_block_sptr> hier_block2_detail::all_blocks()
+{
+    std::set<basic_block_sptr> found_blocks;
+    std::deque<hier_block2_sptr> queued_blocks;
+
+    // Get the blocks from our own flow graph. These can contain hier blocks
+    auto calculated_blocks = d_fg->calc_used_blocks();
+    // Append the internal freestanding blocks
+    for (const auto& freestanding_block : d_blocks) {
+        calculated_blocks.push_back(freestanding_block);
+    }
+
+    for (const auto& block : calculated_blocks) {
+        d_logger->trace("all_blocks: Encountered block {} in calculated_blocks",
+                        block->alias());
+        // Try inserting in the blocks we've found, regardless of whether hier or not
+        if (!found_blocks.insert(block).second) {
+            d_logger->trace("all_blocks: Block {} from calculated_blocks already found",
+                            block->alias());
+            // if the insertion failed, the elemement was already found -> move on
+            continue;
+        }
+
+        // Check for hier-ness of block; if hier, enqueue for later iteration
+        auto hier_block = std::dynamic_pointer_cast<hier_block2>(block);
+        if (hier_block) {
+            d_logger->trace(
+                "all_blocks: Block {} from calculated_blocks is hier. Enqueue.",
+                block->alias());
+            queued_blocks.push_back(hier_block);
+        }
+    }
+
+    while (!queued_blocks.empty()) {
+        const auto dequeed_block = queued_blocks.front();
```

Comment:
Perhaps `dequeued_block` is better than `dequeed_block`?

Reply:
much better!